core/sql/executor/ExHbaseAccess.cpp

```diff
@@ -1645,12 +1650,18 @@ Lng32 ExHbaseAccessTcb::createSQRowFromAlignedFormat(Int64 *latestRowTimestamp)
   
   if (convertExpr())
     {
+      convertRowLen_ = hbaseAccessTdb().convertRowLen();
+      UInt32 rowLen = convertRowLen_;
       ex_expr::exp_return_type evalRetCode =
-	convertExpr()->eval(pentry_down->getAtp(), workAtp_, NULL, asciiRowLen);
+	convertExpr()->eval(pentry_down->getAtp(), workAtp_, NULL, 
+			    asciiRowLen, &rowLen);
       if (evalRetCode == ex_expr::EXPR_ERROR)
 	{
 	  return -1;
 	}
+      if (hbaseAccessTdb().getUseCif() &&
+      rowLen < convertRowLen_)
+        convertRowLen_=rowLen;
```

Comment:
Yes rowLen can be hbaseAccessTdb().convertRowLen(). Row buffer has been pre-allocated with hbaseAccessTdb().convertRowLen(). Now that we have set convertRowLen_ to be actual size of output row for this particular row, I think this ensure that we copy only the required bytes for this output row. Please see change ExHbaseSelect.cpp